esy-lib/Checksum.ml

```diff
@@ -66,12 +66,19 @@ let checkFile ~path (checksum : t) =
       | Sha256, _ -> sha256sum
       | Sha512, _ -> sha512sum
     in
-    let%bind out = ChildProcess.runOut Cmd.(cmd % p path) in
-    match Astring.String.cut ~sep:" " out with
-    | Some (v, _) -> return v
-    | None -> return (String.trim out)
+    (* On Windows, the checksum tools packaged with Cygwin require cygwin-style paths *)
+    let normalizedPath = EsyBash.normalizePathForCygwin (Path.to_string path) in
```

Comment:
I tried using `let%bind` here for both `EsyBash` calls - but unfortunately there is a conflict between the monadic bindings. We have `RunAsync.Syntax` open, but the `EsyBash` utils are using `Result.Syntax`. I'm sure there is a cleaner way to bridge these, as opposed to having to trickle these `match`/`Ok`/`Errors` down - let me know if you have any ideas, @andreypopp .

Reply:
you could locally open different `Syntax` module:
```
let open RunAsync.Syntax in
let%bind async = RunAsync.ofRun (
  let open Run.Syntax in
  let%bind nonasync = ... in
  return ...
) in
return ...
```

Reply:
Cool! Thanks for the suggestion @andreypopp